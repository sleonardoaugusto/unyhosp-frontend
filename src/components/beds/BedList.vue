<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import BedCard from '@/components/beds/BedCard'
import Bedservice from '@/services/Bedservice'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    Bedservice.get()
      .then((res) => {
        this.beds = res.data
      })
  }
}
</script>

<style scoped>

</style>