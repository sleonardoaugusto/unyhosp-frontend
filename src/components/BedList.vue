<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import beds from '@/mocks/beds'
import BedCard from '@/components/BedCard'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    this.beds = beds.data
  }
}
</script>

<style scoped>

</style>