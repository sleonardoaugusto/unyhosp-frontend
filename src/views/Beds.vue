<template>
  <div class="col-md-12">
    <div class="row">
      <div class="col-md-12 d-flex justify-content-center">
        <h1 class="type-h3 type-blue-smoke">Leitos</h1>
      </div>
    </div>
    <div v-if="showList" class="row">
      <bed-list/>
    </div>
    <router-view/>
  </div>
</template>

<script>
import BedList from '@/components/beds/BedList'

export default {
  name: 'home',
  components: {
    BedList
  },
  computed: {
    showList() {
      return this.$route.name === 'beds'
    }
  }
}
</script>
