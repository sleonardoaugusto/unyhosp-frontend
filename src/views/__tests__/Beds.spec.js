import { shallowMount, createLocalVue } from '@vue/test-utils'
import Beds from '@/views/Beds'
import BedList from '@/components/beds/BedList'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('Beds.vue', () => {
  let wrapper

  wrapper = shallowMount(Beds, {
    localVue,
    router
  })

  it('renders component properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders child component', () => {
    expect(wrapper.find(BedList)).toBeTruthy()
  })
})