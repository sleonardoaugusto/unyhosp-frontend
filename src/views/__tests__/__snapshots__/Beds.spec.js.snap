// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beds.vue renders component properly 1`] = `
<div class="col-md-12">
  <div class="row">
    <div class="col-md-12 d-flex justify-content-center">
      <h1 class="type-h3 type-blue-smoke">Leitos</h1>
    </div>
  </div>
  <!---->
  <router-view-stub name="default"></router-view-stub>
</div>
`;
